### PR TITLE
Rule: Find usages of forEach on Ranges

### PR DESCRIPTION
**Pull request by Mauin**
Adds a rule as described in #155 to find usages of forEach on ranges.

I'm still confused that the PSI doesn't allow us to fetch the "parent" of a method which would allow us to check if the parent class is in the `kotlin.ranges` package. Now we parse the receiver expression with a RegEx to verify it matches a range.

### CODE REVIEW

**Comment by arturbosch** [comment]
I think it is a good solution for not having the symbol resolution. I guess the regex would also not interfere with any other strings. :+1:

**Comment by arturbosch** [comment]
@psi - I also found out that most complex operations on psi elements are in some helper classes etc :) http://www.jetbrains.org/intellij/sdk/docs/basics/psi_cookbook.html

**Comment by Mauin** [comment]
I know about `PsiTreeUtil.getParentOfType(...)` which works like a charm for me in Java but I cannot get it to work properly in Kotlin and only get `null` as a result. I'll try to figure that out, matching packages + full classnames in this rule (and also in other rules) should be a lot safer.

**Comment by arturbosch** [comment]
Ahhh, forgot to click `apply review` ... I hate that !

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.jetbrains.kotlin.psi.KtSimpleNameExpression
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import org.jetbrains.kotlin.psi.psiUtil.getCallNameExpression
+import org.jetbrains.kotlin.psi.psiUtil.getReceiverExpression
+
+
+class ForEachOnRange(config: Config = Config.empty) : Rule(config) {
+	override val issue = Issue("ForEachOnRange",
+			Severity.Performance,
+			"Using the forEach method on ranges has a heavy performance cost. Prefer using simple for loops")
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt]: Well maybe not, but later in the Wiki ^^

Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt]: It makes most sense in the Wiki I suppose.

Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt]: Ok, should I just merge this now into master and we try another solution when Resolution rdz?

**Comment by tagantroy** [comment]
@arturbosch Hello, when it will be merged? this PR block #167(Severity.Performance related things)

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt (hunk 1 above)
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/performance/ForEachOnRange.kt]: @arturbosch Yep, I'd say we can merge this as it is and iterate on it later when we have the Context.

**Pull request merged**